src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorUpdateAos.java

```diff
@@ -0,0 +1,86 @@
+import uk.gov.hmcts.divorce.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.common.model.CaseData;
+import uk.gov.hmcts.divorce.common.model.State;
+import uk.gov.hmcts.divorce.common.model.UserRole;
+import uk.gov.hmcts.divorce.document.model.DivorceDocument;
+import uk.gov.hmcts.divorce.solicitor.event.page.Applicant2SolAosCosts;
+import uk.gov.hmcts.divorce.solicitor.event.page.Applicant2SolAosOtherProceedings;
+import uk.gov.hmcts.divorce.solicitor.event.page.Applicant2SolAosjurisdiction;
+import uk.gov.hmcts.divorce.solicitor.event.page.Applicant2SolUpdateAosApplicant1Application;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+import static uk.gov.hmcts.divorce.common.model.State.AosDrafted;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_COURTADMIN_CTSC;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_COURTADMIN_RDU;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_SUPERUSER;
+import static uk.gov.hmcts.divorce.common.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.READ;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.DIVORCE_APPLICATION;
+
+@Component
+public class SolicitorUpdateAos implements CCDConfig<CaseData, State, UserRole> {
+    public static final String SOLICITOR_UPDATE_AOS = "solicitor-update-aos";
+
+    private final List<CcdPageConfiguration> pages = asList(
+        new Applicant2SolUpdateAosApplicant1Application(),
+        new Applicant2SolAosjurisdiction(),
+        new Applicant2SolAosOtherProceedings(),
+        new Applicant2SolAosCosts()
+    );
+
+    @Override
+    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    private PageBuilder addEventConfig(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
+            .event(SOLICITOR_UPDATE_AOS)
+            .forState(AosDrafted)
+            .name("Update AoS")
+            .description("Update Acknowledgement of Service")
+            .aboutToStartCallback(this::aboutToStart)
+            .showSummary()
+            .endButtonLabel("Save Updated AOS Response")
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE, SOLICITOR)
+            .grant(READ,
+                CASEWORKER_COURTADMIN_CTSC,
+                CASEWORKER_COURTADMIN_RDU,
+                CASEWORKER_SUPERUSER,
+                CASEWORKER_LEGAL_ADVISOR));
+    }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<CaseData, State> details) {
```

Comment:
Is this the same logic in the initial create event? If so perhaps it should be shared somehow?

Reply:
Moved common code to AddMiniApplicationLink class